src/Platforms/CUDA/cudaError.h

```diff
@@ -21,10 +21,19 @@
 
 #define cudaErrorCheck(ans, cause)                \
   {                                               \
-    cudaAssert((ans), cause, __FILE__, __LINE__); \
+    cudaAssert((ans), __FILE__, __LINE__, cause);	  \
   }
+
+// The cause is largely redundant with the __FILE__ and __LINE__ information
+// and it makes CUDA heavy code tedious to write and read
+#define cudaCheck(ans)                            \
+  {                                               \
+    cudaAssert((ans), __FILE__, __LINE__); \
+  }                                               \
+
+
 /// prints CUDA error messages. Always use cudaErrorCheck macro.
-inline void cudaAssert(cudaError_t code, const std::string& cause, const char* filename, int line, bool abort = true)
+inline void cudaAssert(cudaError_t code, const std::string& cause, int line, const char* filename = "")
```

Comment:
function arguments are so strange compared to the caller side. There is no need to change this function.
```
cudaErrorCheck(xxx, "")
```
if you prefer not to give additional info via cause. write
```
cudaCheck(ans)
{
cudaAssert((ans),"",  __FILE__, __LINE__); 
}
```

Reply:
You are right, with the macro there is no reason to do this by defaulting the arguement.